web-ng/src/app/components/drawing-kit/drawing-kit.component.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, ChangeDetectionStrategy } from '@angular/core';
+import {
+  DrawingKitService,
+  EditMode,
+} from '../../services/drawing-kit/drawing-kit.service';
+import { ProjectService } from '../../services/project/project.service';
+import { Observable } from 'rxjs';
+import { Layer } from '../../shared/models/layer.model';
+import { List } from 'immutable';
+import { map } from 'rxjs/internal/operators/map';
+import { getPinImageSource } from '../map/ground-pin';
+import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
+
+@Component({
+  selector: 'drawing-kit',
+  templateUrl: './drawing-kit.component.html',
+  styleUrls: ['./drawing-kit.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DrawingKitComponent {
```

Comment:
It's not immediately obvious what "drawing kit" and related service refers to; is it all the drawing tools, the drop down, or the toolbar? If we can choose a more specific name let's do that, if not please add JSDoc to clarify.